pkg/generate/app/cmd/newapp.go

```diff
@@ -662,7 +662,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 			switch {
 			case refInput.ExpectToBuild:
 				glog.V(4).Infof("will add %q secrets into a build for a source build of %q", strings.Join(c.Secrets, ","), refInput.Uses)
-				if err := refInput.Uses.AddBuildSecrets(c.Secrets); err != nil {
+				if err := refInput.Uses.AddBuildSecrets(c.Strategy, c.Secrets); err != nil {
```

Comment:
i don't remember if that is the case or not, i'm afraid you'll have to check unless @csrwng can save you.  everyone should get to dig through new-app once in a while :)